src/reject.rs

```diff
@@ -506,6 +506,13 @@ pub struct MissingHeader {
     name: &'static str,
 }
 
+impl MissingHeader {
+    /// Retrieve the name of the header that was missing
+    pub fn name(self) -> &'static str {
```

Comment:
I'd suggest this should just return `&str`, in case we ever needed to store a dynamic `String` in some cases.

Reply:
Thanks, I was a bit unsure about this lifetime, but I have changed it as suggested to live as long as the rejection itself (I think). Hope this works!

Reply:
Oh, I didn't notice that these methods take `self` by value instead of by reference (`&self`). That should be changed, otherwise you can't call these methods from a `&MissingHeader` etc.

With that change, you don't need to specify any specific lifetime. They'll just look like `pub fn name(&self) -> &str {`.

Reply:
Oh sorry again, I actually didn't know you could do that for some reason. Anyway have hopefully fixed it now and have squashed my commits.